src/arrayMethodJoin.js

```diff
@@ -5,7 +5,16 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const sep = separator === undefined
+      ? ',' : separator === null ? 'null' : separator.toString();
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      str = str + sep + (this[i] === null || this[i] === undefined
+        ? '' : this[i]);
```

Comment:
```suggestion
    if(this[i] != null ) { // non-strict equality for null and undefined
      str += sep + this[i]);
    }  
```

You can use ifs, but don't nest them unnecessarily
Also, checklist point 3

Reply:
Hey, you are wrong in your suggested code it miss value, should put empty string... .
![image](https://github.com/mate-academy/js_array-method-join/assets/134659193/5b34754e-920e-431e-abc6-c7114b5ab36e)

Reply:
You can try this:

``` js 
	if (this[i] != null) {
	  joinString += separator + this[i];
	  continue;
	}
	joinString += separator + "";
```

Reply:
> 

works fine :) thanks and updated :)